week1/project/ecommerce/src/App.jsx

```diff
@@ -0,0 +1,24 @@
+import { useState } from 'react'
+import './App.css'
+import ButtonSet from './ButtonSet.jsx';
+import CardSetMain from './CardSetMain.jsx';
+import allProducts from './fake-data/all-products.js';
+
+function App() {
+  const [filterProducts, setFilterProducts] = useState(allProducts);
+
+  const handleFilterProducts = (category) => {
+    const filtered = allProducts.filter(product => product.category === category.replace(/^FAKE:\s*/, ''));
+    setFilterProducts(filtered);
+  };
+
+  return (
+    <main>
+    <h1>Products</h1>
+    <ButtonSet setFilterProducts={handleFilterProducts} />
```

Comment:
```suggestion
    <ButtonSet onCategorySelect={handleCategorySelect} />
```

**Suggestion:**
For general naming in React, props that handle events should start with 'on'.
1. The 'set' prefix is typically reserved for useState hook naming
2. The name should describe what the prop does, not how it does it
[Reference](https://react.dev/learn/responding-to-events#naming-event-handler-props:~:text=By%20convention%2C%20event%20handler%20props%20should%20start%20with%20on%2C%20followed%20by%20a%20capital%20letter.)